### ISSUE TITLE
Incubating `InclusiveRepositoryContentDescriptor.includeGroupAndSubgroups` breaks dependency resolution

### ISSUE THREAD

**Issue by SimonMarquis** [gradle/gradle#26569]
### Current Behavior

When adding an new repository with `includeGroupAndSubgroups` filtering, the build breaks on basic dependency resolution (even though it does not match anything):

```diff
 repositories {
     mavenCentral()
+    exclusiveContent {
+        forRepositories(maven("https://example.org"))
+        filter {
+            includeGroupAndSubgroups("foo.bar")
+        }
+    }
 }
```

Output of the sample project running `./gradlew build --no-configuration-cache --refresh-dependencies`:

```
> Task :compileKotlin FAILED
1 actionable task: 1 executed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileKotlin'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not find org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.9.0.
     Required by:
         project :

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 31s
```

### Expected Behavior

This new configuration should not break the build.

### Steps to Reproduce

https://github.com/SimonMarquis/Gradle-26569

### Gradle version

8.3

### Your Environment (optional)

Currently tested on Apple M1 Pro, running Ventura 13.5.2.

**Comment by SimonMarquis** [comment]
Looking at the `SimpleSpecMatcher` implementation, everything looks normal, and the [DefaultRepositoryContentDescriptorTest.groovy](https://github.com/gradle/gradle/blob/master/subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/repositories/DefaultRepositoryContentDescriptorTest.groovy) as well.

https://github.com/gradle/gradle/blob/dfc6a93731f21f60e333dc96a9e20730449730ec/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultRepositoryContentDescriptor.java#L411-L421

The issue might probably come from the repositories management.

**Comment by jbartok** [comment]
Thank you for providing a valid report.

The issue is in the backlog of the relevant team, but this area of Gradle is currently not a focus one, so it might take a while before a fix is made.

**Comment by Prototik** [comment]
I think the cause of this is [here](https://github.com/gradle/gradle/blob/73753cc2616266da8d370f6d19fd25d1b9487543/platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/DefaultRepositoryHandler.java#L202)

Note all other methods do invert logic (exclude...), but includeGroupAndSubgroups does direct call, without inverting it to smth like (excludeGroupAndSubgroups)

**Comment by octylFractal** [comment]
Fixed by #27828
